docs/modules/ROOT/pages/guides/v3-migration/index.adoc

```diff
@@ -0,0 +1,159 @@
+[[v3-migration]]
+= 3.0.0 Migration
+This document lists all breaking changes from version 2.x.y to 3.0.0 and how to update.
+
+== How to upgrade
+Simply update `@neo4j/graphql` using npm or your package manager of choice:
+
+[source, bash, indent=0]
+----
+npm update @neo4j/graphql
+----
+
+== Async schema generation
+Schema generation is now asynchronous. Instead of using the property `schema`, now the method `getSchema` will return the schema
+as a _Promise_. This means that creating a server now requires awaiting for that method:
+
+Instead of
+[source, JavaScript, indent=0]
+----
+const neoSchema = new Neo4jGraphQL({ typeDefs, driver });
+const server = new ApolloServer({
+    schema: neoSchema.schema,
+});
+----
+
+Now you'll need to do the following:
+
+[source, JavaScript, indent=0]
+----
+const neoSchema = new Neo4jGraphQL({ typeDefs, driver });
+const schema = await neoSchema.getSchema();
+const server = new ApolloServer({
+    schema: schema
+});
+----
+
+== Relationship changes
+This release contains an overhaul of our relationship validations, this will require a few changes to the schema.
+
+=== Many to * relationships
+To improve consistency and validation, **all** "many to *" relationships need to be defined as _required_ in the schema:
+
+[source, graphql, indent=0]
+----
+type Movie {
+    actors: [Actor!]! @relationship(type: "ACTED_IN", direction: IN)
```

Comment:
What will be the API behaviour if Movie does not have any relationships to Actors? I'm hoping this change doesn't mean there'll be an error.

Reply:
Behaviour will be identical - this always returned an empty List, and this new type definition constraint matches the possible return values.